lib/assert.js

```diff
@@ -106,13 +107,23 @@ function fail(actual, expected, message, operator, stackStartFn) {
       operator = '!=';
   }
 
-  innerFail({
+  if (message instanceof Error) throw message;
+
+  const errArgs = {
     actual,
     expected,
-    message,
-    operator,
+    operator: operator || 'fail',
```

Comment:
Does Node have a consistent way of checking for optional params or default params?
In this case the falsey check means that users can't specify it as `null` or `""`.

Reply:
No, it's often mixed and fasly values are often set to a default while in other cases only undefined is set to a default.

Reply:
This method has a mix of deprecated/legacy and new/different functionality. Would moving the operator juggle into a `argLength < 2` branch with the other `argLen` checks be doable? That way it touches less of the overlapping code paths.

Reply:
That was done on purpose since I believe it is best to just properly do this for all cases.

Shall I maybe change the check to: `operator === undefined ? 'fail' : operator`?